File: Server/src/logic/player/Board.java
package logic.player;

import logic.cards.Permanent;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by Aedan Smith.
 */

public interface Board {
    void add(Permanent permanent);

    String remove(int n);

    List<Permanent> getBoard();

    default void applyStateBasedActions(){
        ArrayList<Permanent> toRemove = new ArrayList<>();
        for (Permanent permanent : getBoard()) {
            if (permanent.health <= 0) {
                toRemove.add(permanent);
            }
        }
        for (Permanent permanent : toRemove) {
            remove(getBoard().indexOf(permanent));
        }
    }
}

File: Server/src/logic/player/Board.kt
package logic.player

import logic.cards.Permanent
import java.util.*

/**
 * Created by Aedan Smith.
 */

interface Board {
    fun add(permanent: Permanent)

    fun remove(n: Int): String

    val board: List<Permanent>

    fun applyStateBasedActions() {
        val toRemove = ArrayList<Permanent>()
        for (permanent in board) {
            if (permanent.health <= 0) {
                toRemove.add(permanent)
            }
        }
        for (permanent in toRemove) {
            remove(board.indexOf(permanent))
        }
    }
}
